src/const_eval.md

```diff
@@ -67,6 +67,30 @@ surrounding generic parameters: such an expression must either be a single bare
 const generic parameter, or an arbitrary expression not making use of any
 generics.
 
+## Permitted use of `static`s
+
+Within the constant evaluation context, items that ultimately contain no mutable memory or data that is
```

Comment:
I'm struggling to follow this section. It seems like you are allowed to borrow static mut items, as in:

```rust
const fn f() -> &'static u32 {
    static mut VALUE: u32 = 0;
    unsafe { &VALUE }
}

fn main() {
    let y = const { f() };
}
```

I can't really tell why it is not allowed in the initializer of a `const` as shown below. What is the rule there, and why is it different?

(I'm probably very confused, but the example above also seems to go against what the stabilization report says.)

Additionally, the stabilization report seems to make a distinction of *resulting* in a value that contains a reference to something mutable. (As opposed to merely accessing mutable statics within the course of evaluation, which seems to be allowed?)

Reply:
> It seems like you are allowed to borrow static mut items, as in:

Yes, but you can't read from them. (It's also questionable whether this may be UB, but probably not worth explicitly making UB.)

> I can't really tell why it is not allowed in the initializer of a const as shown below. What is the rule there, and why is it different?

It is not allowed if the reference ends up in the final value of the const *as a safely readable reference*. (If it ends up as a raw pointer, or just inside a union, it's fine.) This is because such consts can be used as patterns, and patterns depending on mutable data would be Bad News.